mdal/mdal_logger.hpp

```diff
@@ -12,17 +12,39 @@
 
 namespace MDAL
 {
+  /** Namespace including functions responsible for handling logs.
+   * Use in code as: MDAL::Log::error/warning( MDAL_Status, logMessage ).
+   * By default, output from logger is not being shown anywhere, but it is
+   * possible to set custom logger output with function setLoggerCallback.
+   * If environment variable MDAL_DEBUG is set, logger sends logs to standard stdout. */
   namespace Log
   {
+    /** Logger function handling error logs */
```

Comment:
add comments to the interal functions only when it adds any additional context/help on top of the obvious function name + arg list... otherwise the comment is not required